src/Uno.UI.Toolkit/Storage/StorageFileHelper.Android.cs

```diff
@@ -0,0 +1,101 @@
+﻿#nullable enable
+#if __ANDROID__
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Android.Content.Res;
+using System.Threading.Tasks;
+
+namespace Uno.UI.Toolkit;
+
+partial class StorageFileHelper
+{
+	private static ICollection<string> _scannedFiles { get; set; } = new List<string>();
+
+	private static Task<bool> FileExistsInPackage(string fileName)
+	{
+		//Look in assets first***
+		var context = global::Android.App.Application.Context;
+
+		//Get only the filename with extension and apply internal UNO naming rules so we are able to find the compiled resources by replacing '.' and '-' by underscore.
+		var normalizedFileName = Path.GetFileNameWithoutExtension(fileName).Replace('.', '_').Replace('-', '_') + Path.GetExtension(fileName);
+		if (!_scannedFiles.Any())
+		{
+			ScanPackageAssets(_scannedFiles);
+		}
+
+		if (_scannedFiles.Contains(normalizedFileName))
+		{
+			return Task.FromResult(true);
+		}
+
+		//If not asset found and we detect "/" in filename means we are looking for a nested resource file.
+		//In this case we need to replace '/, -, .' by '_' and remove filename extension before trying to pull it from app resources.
+		var normalizedResName = fileName.ToLowerInvariant();
+		var nameArray = normalizedResName.Split("/")?.ToList();
```

Comment:
split should never be returning null. I also think it's not necessary to convert to list?

```suggestion
		var nameArray = normalizedResName.Split("/");
```